es_find_block.py

```diff
@@ -0,0 +1,45 @@
+from .es_base_block import ESBase, Limitable, Sortable
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.metadata.properties.expression import ExpressionProperty
+from nio.common.signal.base import Signal
+
+
+@Discoverable(DiscoverableType.block)
+class ESFind(Limitable, Sortable, ESBase):
+
+    """ A block for running `search` against a elasticsearch.
+
+    Properties:
+        condition (expression): A dictionary form of a search expression.
+        This is an expression property that can evaluate to a dictionary
+        or be a parseable JSON string
+
+    """
+    condition = ExpressionProperty(title='Condition',
+                                   default="{'match_all': {}}")
+
+    def execute_query(self, doc_type, signal):
+        condition = self.evaluate_expression(self.condition, signal)
+        self._logger.debug("Condition evaluated to: {}".format(condition))
+
+        search_results = self.search(doc_type=doc_type,
+                                     body={"query": condition},
+                                     params=self.query_args())
+        if search_results and "hits" in search_results:
+            return [Signal(self._process_fields(hit))
```

Comment:
The only problem with this is that the result is going to be buried in the `$source` object of the output signal. There may be times when this is desired but in the normal case the signal should probably just be the source object.

I suggest adding a flag to return the entire query result which is unchecked by default